File: backend/app/schemas/cv.py
from pydantic import BaseModel
from typing import Optional, List


class ContactInfo(BaseModel):
    name: Optional[str] = None
    email: Optional[str] = None
    phone: Optional[str] = None
    location: Optional[str] = None
    linkedin: Optional[str] = None
    website: Optional[str] = None


class Experience(BaseModel):
    job_title: str
    company: str
    location: Optional[str] = None
    start_date: Optional[str] = None
    end_date: Optional[str] = None
    current: bool = False
    bullets: List[str] = []


class Education(BaseModel):
    school: str
    degree: str
    field: Optional[str] = None
    graduation_date: Optional[str] = None
    gpa: Optional[str] = None


class CVContent(BaseModel):
    contact_info: Optional[ContactInfo] = None
    summary: Optional[str] = None
    experience: List[Experience] = []
    education: List[Education] = []
    skills: List[str] = []
    certifications: List[str] = []
    languages: List[str] = []
    projects: List[dict] = []
    publications: List[dict] = []
    volunteer: List[dict] = []
    custom_sections: List[dict] = []


class CVCreate(BaseModel):
    title: str = "Untitled CV"
    template_id: str = "olive"


class CVUpdate(BaseModel):
    title: Optional[str] = None
    template_id: Optional[str] = None
    content: Optional[CVContent] = None


class CVSectionUpdate(BaseModel):
    section: str
    data: dict


class CVSummary(BaseModel):
    id: str
    title: str
    template_id: str
    ats_score: Optional[int] = None
    status: str = "draft"
    created_at: str
    updated_at: str


class CVDetail(BaseModel):
    id: str
    title: str
    template_id: str
    content: CVContent
    ats_score: Optional[int] = None
    status: str = "draft"
    created_at: str
    updated_at: str

File: frontend/lib/api/cv.ts
import { request, download } from "./client";

export interface CVSummary {
  id: string;
  title: string;
  template_id: string;
  ats_score: number | null;
  status: "draft" | "complete";
  created_at: string;
  updated_at: string;
}

export interface CVContent {
  contact_info?: {
    name?: string;
    email?: string;
    phone?: string;
    location?: string;
    linkedin?: string;
    website?: string;
  };
  summary?: string;
  experience?: Array<{
    job_title: string;
    company: string;
    location?: string;
    start_date?: string;
    end_date?: string;
    current?: boolean;
    bullets: string[];
  }>;
  education?: Array<{
    school: string;
    degree: string;
    field?: string;
    graduation_date?: string;
    gpa?: string;
  }>;
  skills?: string[];
  certifications?: string[];
  languages?: string[];
  projects?: Record<string, unknown>[];
  publications?: Record<string, unknown>[];
  volunteer?: Record<string, unknown>[];
  custom_sections?: Record<string, unknown>[];
}

export interface CVDetail extends CVSummary {
  content: CVContent;
}

export const cvApi = {
  list: () => request<CVSummary[]>("/cv/"),

  get: (id: string) => request<CVDetail>(`/cv/${id}`),

  create: (title = "Untitled CV", templateId = "olive") =>
    request<CVDetail>("/cv/", {
      method: "POST",
      body: JSON.stringify({ title, template_id: templateId }),
    }),

  update: (id: string, data: { title?: string; template_id?: string; content?: CVContent }) =>
    request<CVDetail>(`/cv/${id}`, {
      method: "PUT",
      body: JSON.stringify(data),
    }),

  autoSave: (id: string, content: CVContent) =>
    request<{ saved: boolean }>(`/cv/${id}/auto-save`, {
      method: "POST",
      body: JSON.stringify({ content }),
    }),

  duplicate: (id: string) =>
    request<CVDetail>(`/cv/${id}/duplicate`, { method: "POST" }),

  delete: (id: string) =>
    request<void>(`/cv/${id}`, { method: "DELETE" }),

  downloadPreview: (id: string) =>
    download(`/cv/${id}/preview`),
};

export const cvAiApi = {
  improveText: (text: string, context: string, language = "en") =>
    request<{ improved_text: string }>("/cv/ai/improve-text", {
      method: "POST",
      body: JSON.stringify({ text, context, language }),
    }),

  generateSummary: (cvId: string, targetRole: string, language = "en") =>
    request<{ summary: string }>("/cv/ai/generate-summary", {
      method: "POST",
      body: JSON.stringify({ cv_id: cvId, target_role: targetRole, language }),
    }),

  suggestBullets: (jobTitle: string, company?: string, industry?: string) =>
    request<{ bullets: string[] }>("/cv/ai/suggest-bullets", {
      method: "POST",
      body: JSON.stringify({ job_title: jobTitle, company, industry }),
    }),
};
